### PR TITLE
V14: Server information

### PR DESCRIPTION
**Pull request by Zeegaan**
# Notes
- Refactors `Server/Information` endpoint to `Server/Troubleshooting`
- Adds new `Server/Information` endpoint
- Add new method to ISystemInformationService, to get out this new server information.
- Implemented `ServerConfigurationBaseModel`, as we're going to implement many more endpoints that will return structured configuration like this.

# How to test
- Using swagger
- Test the `Server/Troubleshooting`, and assert it still return the same as `Server/Information` did before
- Test the `Server/Information endpoint`, this should return something like it these in the old server variables:
```
       "version": "14.0.0-preview004.preview.221",
        "assemblyVersion": "14.0.0.0",
        "serverTimeOffset": 60
        "runtimeMode": "BackofficeDevelopment",
```

### CODE REVIEW

File: src/Umbraco.Cms.Api.Management/Controllers/Server/InformationServerController.cs [hunk 1]
```diff
@@ -11,21 +9,14 @@ namespace Umbraco.Cms.Api.Management.Controllers.Server;
...
     [HttpGet("information")]
     [MapToApiVersion("1.0")]
-    [ProducesResponseType(typeof(ServerInformationResponseModel), StatusCodes.Status200OK)]
-    public Task<IActionResult> GetServerInformation()
-    {
-        ServerInformationResponseModel responseModel = _mapper.Map<ServerInformationResponseModel>(_systemInformationService.GetSystemInformation())!;
-
-        return Task.FromResult<IActionResult>(Ok(responseModel));
-    }
+    [ProducesResponseType(typeof(Dictionary<string, object>), StatusCodes.Status200OK)]
```
Comment (Zeegaan) [inline src/Umbraco.Cms.Api.Management/Controllers/Server/InformationServerController.cs:20]: Seems wierd to not use a viewmodel for this, not entirely sure if this is the best approach, or if we should switch to a proper ViewModel

Comment (bergmania) [reply src/Umbraco.Cms.Api.Management/Controllers/Server/InformationServerController.cs]: When we return a ServerInformationResponseModel it should be that model

File: src/Umbraco.Infrastructure/Telemetry/Providers/SystemInformationTelemetryProvider.cs [hunk 2]
```diff
@@ -111,6 +111,15 @@ internal class SystemInformationTelemetryProvider : IDetailedTelemetryProvider,
             { "Current Server Role", CurrentServerRole },
         };
 
+    public IDictionary<string, object> GetServerInformation() =>
```
Comment (bergmania) [inline src/Umbraco.Infrastructure/Telemetry/Providers/SystemInformationTelemetryProvider.cs:114]: IMO Serverinformation is different from Troubleshooting info. I think we should have an explicit model for server information

File: tests/Umbraco.Tests.UnitTests/Umbraco.Core/Services/SystemInformationServiceTests.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ public class SystemInformationServiceTests
     public void GetCorrectDefaultLanguageTest(string culture)
     {
         var userDataService = CreateSystemInformationService(culture);
-        var defaultLanguage = userDataService.GetSystemInformation().FirstOrDefault(x => x.Key == "Default Language");
```
Comment (bergmania) [inline tests/Umbraco.Tests.UnitTests/Umbraco.Core/Services/SystemInformationServiceTests.cs:33]: This is also nasty without an explicit model

**Review by bergmania — CHANGES_REQUESTED** [review]
A couple of comments.. Let me know if I should just fix it up

File: src/Umbraco.Infrastructure/Telemetry/Providers/SystemInformationTelemetryProvider.cs [hunk 4]
```diff
@@ -111,6 +111,15 @@ internal class SystemInformationTelemetryProvider : IDetailedTelemetryProvider,
...
+    new Dictionary<string, object>
+    {
+        { "version", _version.SemanticVersion.ToSemanticString() },
+        { "assemblyVersion", _version.SemanticVersion.ToSemanticStringWithoutBuild() },
+        { "serverTimeOffset", Convert.ToInt32(DateTimeOffset.Now.Offset.TotalMinutes) },
```
Comment (bergmania) [inline src/Umbraco.Infrastructure/Telemetry/Providers/SystemInformationTelemetryProvider.cs:119]: Also never use DateTimeOffset.Now, inject TimeProvider instead

**Pull request merged**